### DISCUSSION TITLE
Welcome to tsParticles general discussion!

### DISCUSSION THREAD

**Discussion by matteobruni** [matteobruni/tsparticles#1114] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by ohmygaugh-crypto** [comment]
Seasons Greetings @matteobruni! First of all thanks for starting up this really cool repo. I hope to contribute more as a get better at programming! Anyways, I'm very curious about the polygon mask feature and wish to expand my creative ability and bug fixing with said feature. So, with that being said I do have a question for you and I will try to keep it brief:    Regarding some of your codepen template(s) (such as the HollowKnight)... how are you sooo successful at "tracing between the lines"? The particles have very clear a defined paths/boundaries that I haven't been able to replicate for my use cases.  Which properties would you suggest I address within my own code? Let me know if you need me to submit an example of my code for reference? Cheers💪!

**Reply by matteobruni** [reply]
Hi @ohmygaugh-crypto,

The polygon mask seems easy but sometimes it can bring some issues that can be annoying. To have a good perimeter around the lines you need to have closer particles, so if the polygon is big you need a high number of particles, and this could bring some performance issues. With closer particles, one another improvement is closer links distance because they are very expensive and it's not nice having particles linking between SVG lines.

If you can provide a reproducible sample we can look for some config improvements.

**Reply by ohmygaugh-crypto** [reply]
Hey actually you are spot on! I used an svg converter website to shrink the size ratio of the pathways(by 1/2), so I could implement wayyyy less particles. Additionally, I decreased the: "polygon{ move: { radius: from 30 --> to 5 }}" which tightened up the particles. I then slightly bumped up the particle number value, while decreasing the link distance. Now I just have to figure out how you constructed  that  italian flag template, so the colors are split in diagonals rather than vertical. I can provide documentation soon for others facing this soon. Thanks for the promptness! Let me get back to you If I can't figure out the Italian flag template. best

**Reply by ohmygaugh-crypto** [reply]
@matteobruni Can you even separate particle event colors with "diagonal" divs on top of a polygon mask? Have you tried that before?

**Reply by matteobruni** [reply]
I never tried the diagonal div but I don't think it works since there are only two shapes handled: rectangle and circle.

About color separation on top of a polygon mask I never tried but it should work (I think).

**Reply by ohmygaugh-crypto** [reply]
@matteobruni here is what I have thus far with my project. Still a work in progress b/c there are two main features left to add. I'll post the final product on the "show-&-tell" channel, but for now, I just wanted to show anyone else who comes across this thread what we are talking about ha: https://codepen.io/ohmygaugh-crypto/pen/vYXxNEv

**Reply by ohmygaugh-crypto** [reply]
> 🍝

**Reply by ohmygaugh-crypto** [reply]
@matteobruni how(what src url) are you pulling in those logo particles logo images? I'm just getting floating rectangles: https://codepen.io/ohmygaugh-crypto/pen/qBamwEJ

**Reply by ohmygaugh-crypto** [reply]
@matteobruni I'm dumb lol. I simply forgot to add the <script> into the HTML

**Comment by stovmascript** [comment]
Hi. I have a question about package size. Compared to particles.js' 22kB (https://bundlephobia.com/result?p=particles.js@2.0.0), tsparticles weighs 170kB (https://bundlephobia.com/result?p=tsparticles@1.18.11). Is it larger because it has more features or am I missing something?

**Reply by matteobruni** [reply]
Yes, the main difference is that.

Another feature that occupies a lot of space is the options compatibility (from `particles.js`, since they have been refactored a lot) but in the v2 version the size will be reviewed and I hope for mode modularity and customizations, if you don't need a feature you can skip it entirely (if you prefer the composable version) but it's a work in progress.

**Comment by ohmygaugh-crypto** [comment]
@matteobruni is SCSS compatible with tsparticles? I wanted to overlay a 3D "Learn More"  button ontop of my design. Is adding this possible? I've been unsuccessful thus far, so I figured I'd reach out again. Also how "neat" does the show and tell need to be? Is it better I break down my project into checkpoints you think?

**Reply by matteobruni** [reply]
You can create your own Show and Tell discussion if you want to show your steps. SCSS won't interfere with tsParticles, you can change the canvas or container CSS as you want.
I don't understand very well what you want to achieve with the button, can you give me more details?

**Comment by angelb2m** [comment]
https://static.iohk.io/fluid2/

Puedo hacer este efecto con TSparticles?

**Reply by matteobruni** [reply]
Hi @angelb2m,

No this effect is still not possible yet but there's an open issue for a similar effect here #208
